src/Http/WebUtilities/src/HttpRequestStreamReader.cs

```diff
@@ -370,5 +370,19 @@ private async Task<int> ReadIntoBufferAsync()
 
             return _charsRead;
         }
+
+        public async override Task<string> ReadToEndAsync()
+        {
+            StringBuilder sb = new StringBuilder(_charsRead - _charBufferIndex);
+            do
+            {
+                int tmpCharPos = _charBufferIndex;
```

Comment:
Why do you need tmpCharPos?

Reply:
Again, this comes from `StreamReader.ReadToEndAsync`. I guess it's taking a copy of the field in case it's modified by another thread. The sync version of the method doesn't take a copy and uses the field directly, though.

Reply:
If it matches the base code, I think we can leave it.